ASSIGMENT/src/pages/voter.astro

```diff
@@ -0,0 +1,270 @@
+    <canvas id="voterCanvas"></canvas>
+    <button id="downloadButton" class="download" style="display: none;">Download Image</button>
+  </div>
+</MainLayout>
+<script is:inline>
+    const voterForm = document.querySelector(".form");
+    const saveButton = document.querySelector(".saveButton");
+    const resetButton = document.querySelector(".resetButton");
+    const container = document.querySelector(".container");
+    const canvasContainer = document.getElementById("canvasContainer");
+    const downloadButton = document.getElementById("downloadButton");
+    const canvas = document.getElementById("voterCanvas");
+    const context = canvas.getContext("2d");
+  
+    let entryCount = 0;
+    let currentEditItem = null;
+  
+    const handleSubmit = (event) => {
+      event.preventDefault();
+  
+      const documentNumber = voterForm.querySelector("#doc-number").value.trim();
+      const holdingPersonName = voterForm.querySelector("#doc-name").value.trim();
+      const fathername = voterForm.querySelector("#doc-fathername").value.trim();
+      const DOB = voterForm.querySelector("#doc-dob").value.trim();
+      const gender = voterForm.querySelector('input[name="gender"]:checked').value;
+  
+      if (documentNumber && holdingPersonName && DOB && fathername) {
+        if (currentEditItem) {
+          updateEntry(currentEditItem, documentNumber, holdingPersonName, fathername, DOB, gender);
+        } else {
+          createNewEntry(documentNumber, holdingPersonName, fathername, DOB, gender);
+        }
+        resetForm();
+      } else {
+        console.log("Please fill in all required fields.");
+      }
+    };
+  
+    const createNewEntry = (documentNumber, holdingPersonName, fathername, DOB, gender) => {
+      entryCount++;
+      const maleSymbol = String.fromCharCode(0x2642);
+      const femaleSymbol = String.fromCharCode(0x2640);
+      const genderSymbol = gender === "male" ? maleSymbol : femaleSymbol;
+  
+      container.insertAdjacentHTML('beforeend', `
+        <div class="item" data-entry-id="${entryCount}" data-address="${fathername}">
+          <div>${entryCount}</div>
+          <div>Voter ID</div>
+          <div>${documentNumber}</div>
+          <div>${holdingPersonName}</div>
+          <div>${genderSymbol}</div>
+          <div>${formatDate(DOB)}</div>
+          <div>
+            <button class="view-btn" type="button">View</button>
+            <button class="edit-btn" type="button">Edit</button>
+            <button class="delete-btn" type="button">Delete</button>
+          </div>
+        </div>
+      `);
+    };
+  
+    const updateEntry = (item, documentNumber, holdingPersonName, fathername, DOB, gender) => {
+      const maleSymbol = String.fromCharCode(0x2642);
+      const femaleSymbol = String.fromCharCode(0x2640);
+      const genderSymbol = gender === "male" ? maleSymbol : femaleSymbol;
+  
+      item.querySelector("div:nth-child(3)").textContent = documentNumber;
```

Comment:
no  `nth child` thing to be done. remove from all places

Reply:
done